src/posix/platform/infra_if.cpp

```diff
@@ -726,15 +771,44 @@ otError InfraNetif::DiscoverNat64Prefix(uint32_t aInfraIfIndex)
 }
 #endif // OPENTHREAD_POSIX_CONFIG_NAT64_AIL_PREFIX_ENABLE
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
+void InfraNetif::SetInfraNetifForBorderRouting(int aIcmp6Socket)
+{
+    otBorderRoutingState state = otBorderRoutingGetState(gInstance);
+
+    VerifyOrDie(state == OT_BORDER_ROUTING_STATE_UNINITIALIZED || state == OT_BORDER_ROUTING_STATE_DISABLED,
+                OT_EXIT_INVALID_STATE);
+
+    if (mInfraIfIcmp6Socket != -1)
+    {
+        close(mInfraIfIcmp6Socket);
+    }
+    mInfraIfIcmp6Socket = aIcmp6Socket;
+}
+#endif // OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
+
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+void InfraNetif::SetInfraNetifForBackboneRouter(void)
```

Comment:
this doesn't seem to do anything, do we still need it?

Reply:
At least the check is still necessary otherwise we may set the infra link at a wrong state. 

It may be arguable whether `SetInfraNetifForBackboneRouter` should be an individual method. IMO it's neater to have it to be on par with `SetInfraNetifForBorderRouting`.

Reply:
The function name doesn't really describe what is done in the function, maybe rename the function or just have the check in `InfraNetif::SetInfraNetif` before setting the `mInfraIfName` and `mInfraIfIndex`.